cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java

```diff
@@ -94,12 +97,62 @@ public void execute() {
                 targetFileName = pkgName;
             }
 
-            BuilderUtils.compileWithTestsAndWrite(sourceRootPath, pkgName, targetFileName, buildCompiledPkg, offline,
-                                                  lockEnabled, skiptests);
+            Path sourcePath = Paths.get(pkgName);
+            Path resolvedFullPath = sourceRootPath.resolve(sourcePath);
+            // If the source is a single bal file which is not inside a project
+            if (Files.isRegularFile(resolvedFullPath) &&
+                    sourcePath.toString().endsWith(BLangConstants.BLANG_SRC_FILE_SUFFIX) &&
+                    !RepoUtils.hasProjectRepo(sourceRootPath)) {
```

Comment:
Whats the difference between source path and source root path?

Reply:
so shall we use correct names for the variables instead?

Reply:
Added proper variable names with 70bc3df